analyses/chromosomal-instability/util/circos-plots.R

```diff
@@ -0,0 +1,338 @@
+# Functions for circos plots
+#
+# C. Savonen for ALSF - CCDL
+#
+# 2020
+#
+prep_bed <- function(df,
+                     samples_col = "samples",
+                     sample_names = "all",
+                     chr_col = "chrom",
+                     start_col = "start",
+                     end_col = "end",
+                     y_val = "none") {
+  # Given a data.frame, set it up for use with circlize. Note that start and end
+  # columns cannot be the same.
+  #
+  # Args:
+  #   df: a data.frame with the chromosomal coordinates and y value to plot
+  #   samples_col: a character string specifying the samples column which can be used to filter by.
+  #   sample_names: a character string that specifies values to keep from `samples_col` column.
+  #                 "all" keeps all samples in.
+  #   chr_col: a character string that specifies the chromosomes column name.
+  #   start_col: a character string that specifies the start coordinate column name.
+  #   end_col: a character string that specifies the end coordinate column name.
+  #   y_val: The column name of the value you would like to plot (Optional)
+  #
+  # Returns:
+  #  Data.frame formatted how circlize wants it
+
+  # Filter the samples specified
+  if (sample_names[1] != "all") {
+    df <- df %>%
+      # Filter based on the samples
+      dplyr::filter(!!rlang::sym(samples_col) %in% sample_names)
+  }
+
+  # Set up the data how circlize wants it
+  bed_df <- df %>%
+    # Pull out the specified columns
+    dplyr::select(
+      chr = !!rlang::sym(chr_col),
+      start = !!rlang::sym(start_col),
+      end = !!rlang::sym(end_col)
+    )
+
+  # Don't have to have a y_val
+  if (y_val != "none") {
+    bed_df <- bed_df %>%
+      dplyr::mutate(y_val = as.numeric(dplyr::pull(df, !!rlang::sym(y_val))))
+  }
+  # Circlize is particular about what these values need to be
+  bed_df <- bed_df %>%
+    dplyr::mutate(
+      chr = as.character(chr),
+      start = as.integer(start),
+      end = as.integer(end)
+    ) %>%
+    # Circlize wants things to be only a data.frame NOT a tibble
+    as.data.frame()
+
+  # Stop if there isn't data in this data.frame
+  if (nrow(bed_df) == 0) {
+    stop("There are 0 rows of data. Check your samples_col arg and the data.frame you supplied. ")
+  }
+
+  # Check if the chromosomes have the `chr` format. If not, add it.
+  if (!grepl("chr", bed_df$chr[1])) {
+    bed_df <- bed_df %>%
+      dplyr::mutate(chr = paste0("chr", chr))
+  }
+
+  # Return formatted data.frame
+  return(bed_df)
+}
+
+circos_map_plot <- function(df,
+                            add_track = FALSE,
+                            samples_col = "samples",
+                            sample_names = NULL,
+                            chr_col = "chrom",
+                            start_col = "start",
+                            end_col = "end",
```

Comment:
All the data file seem to have different column names so this is my way to try to make it flexible. How desirable/advantageous is it to have shorter function calls?

Reply:
It can stay just as flexible. But personally this is a case where I would find myself renaming data frames to match the default column names so the function call itself would be shorter. But definitely don't remove the option!